lib/LaTeXML/Core/Parameters.pm

```diff
@@ -59,6 +60,7 @@ sub revertArguments {
 sub readArguments {
   my ($self, $gullet, $fordefn) = @_;
   my @args = ();
+  $gullet->setup_scan() if $$gullet{pushback_has_smuggled_the};
```

Comment:
how expensive is moving the `if` to Gullet's `setup_scan`?  I'd think not so much, and would avoid the secret knowledge & `$$`.

Reply:
Well, it's north of 4,000 pointless function calls... and it's a lot cheaper to check the hash value than to call *and* check. `readArguments` is called quite often as you can imagine.

Let me get a number from the parallel make test, comparing the two... +2 seconds in the parallel make test. Not the end of the world, my first naive stab *doubled* the time of make test :> But it's such an easy optimization that it's kind of a shame not to do it...

Reply:
... but 2 seconds aren't that big a deal, committed!